### ISSUE TITLE
Negative Sharpness Values

### ISSUE THREAD

**Issue by giorgiopiras** [tml-epfl/sharpness-vs-generalization#2]
Hey @max-andr, thanks for your work on this repo. 

I am using your code on CIFAR10 for a WRN-28-10, and I see that using pretrained models (and thus evaluating the sharpness on the test set), the average sharpness is in some cases negative (I am using 100 directions). 
This means that, on those 100 average random directions, in some cases (most of them, actually) the loss happens to be lower using w+delta. 
I saw that in your code the loss difference is not taken as an absolute value though, so I was wondering whether this could be something that you simply did not experience on your experiments or, maybe, a signal that there is something which is not working correctly in the way I am using the code. 

PS: forgot one detail...this happens more frequently (and reasonably so) for small rho values (e.g., 0.01 to 0.1)...as rho grows (e.g., 0.2-0.6) this tends to be more rare (but still, it happens). 

Thanks a lot! Giorgio

**Comment by max-andr** [comment]
Hey Giorgio,

Yes, I've also observed the same and also only for small rho value. I think it's not too surprising: if rho is small, then the function is approximately constant so some average-case sharpness values are slightly > 0 and some are slightly < 0. 

And note that while rho=0.1 is not too small for worst-case sharpness, it's really tiny for average-case sharpness. I'm not sure if it makes sense to really consider average-case sharpness at such scale (at least, we didn't).

Best,
Maksym

**Comment by giorgiopiras** [comment]
Thanks for the answer, Maksym.

In this regard, I am testing various rho values up to 0.008 for avg_linf and 0.8 for worst case (still Linf); mostly based on your paper's experiments and eval_sharpness_final.sh file on CIFAR10. Do you think that is too low? 
Did you apply any criterion to choose the values on your model/dataset pairs (e.g., some sharpness value you reached without logit normalization that made you think that was a proper rho value)? 

Thanks, 
Giorgio
